src/frontend/src/binder/query.rs

```diff
@@ -77,6 +78,9 @@ impl Binder {
     pub(super) fn bind_query_inner(&mut self, query: Query) -> Result<BoundQuery> {
         let limit = query.get_limit_value();
         let offset = query.get_offset_value();
+        if offset.is_some() && limit.is_none() {
```

Comment:
Only stream queries don't support this. Batch queries can support offset without limit. I think the check could be moved to the place where we generate TopN. If at that optimize stage, we found that there's order by without limit, we can simply return an error.